usr/lib/pkcs11/common/attributes.c

```diff
@@ -22,23 +22,44 @@
 #include "host_defs.h"
 #include "h_extern.h"
 #include "trace.h"
+#include <openssl/crypto.h>
 
-/*
- * Free an array of attributes allocated with dup_attribute_array().
- */
-void free_attribute_array(CK_ATTRIBUTE_PTR attrs, CK_ULONG attrs_len)
+static void __cleanse_and_free_attribute_array(CK_ATTRIBUTE_PTR attrs,
```

Comment:
Whats the reason for no inline (it possibly gets inline ed anyway at higher optimization levels) ?

Reply:
I thought I let this to the compiler, since it will most likely be inlined anyway... If you want me to make in inlined, let me know.

Reply:
Decision is left to compiler anyway: inline keyword is only a suggestion to the compiler, compiler is not required to actually do the inlining ;-)

Not inlining it isokay for me. Apporved.